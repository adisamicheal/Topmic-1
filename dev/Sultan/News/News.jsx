import React, {Component} from 'react';

export default class News extends Component{
	constructor(){
		super();
	}


	render(){
		return (

				<div className="News">
					
					<div className="news-Box">

						<div className="uper-box">

							<h2>Latest News</h2>
							<p> Sed posuere consectetur est at lobor,
								Etiam porta sem malesuada magna..<br/>
								<span>03.28.18</span>
							</p>

							
							<p> Sed posuere consectetur est at lobor,
								Etiam porta sem malesuada magna..<br/>
								<span>03.27.18</span>
							</p>

						</div>


						<div className="lower-box">

							<div className="des-box">
								<h3>Education Blog</h3>
							</div>
							<div className="des-box box-2">
								<h3>Resource center</h3>
							</div>
							<div className="des-box box-3">
								<h3>Winsome Gallery</h3>
							</div>
							<div className="des-box box-4">
								<h3>Annual Achievers</h3>
							</div>

						</div>


					</div>

					<div className="clear">
					</div>


				</div>

			);
	}
}