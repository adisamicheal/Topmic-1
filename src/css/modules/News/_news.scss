//############# color Variables  ##########
$uper-box-color: #404040;
$box-1-color: #ffcc33;
$box-2-color: #ff6666;
$box-3-color: #8080ff;
$box-4-color: #80ff80;
$box-border-color: #404040;
$news-resp-screen: 769px;

//######### SCSS of News ###################
.News {
	.news-Box {
		width: 100%;

		.uper-box {
			width: 100%;
			background-color: $uper-box-color;
			h2 {
				margin: 0px;
				padding: 50px 40px;
				color: white;
				font-family: sans-serif;
				@media only screen and (max-width: 940px) {
					padding: 30px 20px;
					font-size: 24px;
				}
			}
			p {
				@include P(left, white);
				padding: 0px 40px 50px 40px;
				font-size: 18px;
				text-align: justify;
				font-family: sans-serif;
				color: #d9d9d9;
				@media only screen and (max-width: 940px) {
					padding: 0px 20px 30px 20px;
					font-size: 14px;
				}
				span {
					color: $box-1-color;
				}
			}
		}
		.lower-box {
			width: 100%;

			.des-box {
				height: 100px;
				width: 25%;
				background-color: $box-1-color;
				float: left;

				@media only screen and (min-width: $news-resp-screen) {
					height: 80px;
					width: 100%;
				}
				@media only screen and(min-width:940px) {
					height: 50%;
					width: 50%;
				}
				h3 {
					font-family: sans-serif;
					padding: 0 20%;
					margin: 35% 0;
					border-left: 9px $box-border-color solid;
					box-sizing: border-box;
					font-weight: 700;
					@media only screen and (max-width: 1300px) {
						font-weight: 600;
						padding: 0 5%;
						font-size: 18px;
					}
					@media only screen and (max-width: 940px) {
						margin: 35px 0;
					}
					@media only screen and (max-width: $news-resp-screen) {
						font-weight: 600;
						padding: 0 7%;
						font-size: 22px;
						border-left: 5px $box-border-color solid;
					}
					@media only screen and (max-width: 500px) {
						font-weight: 550;
						padding: 0 6%;
						font-size: 15px;
						border-left: 3px $box-border-color solid;
					}
				}
			}
			.box-2 {
				background-color: $box-2-color;
			}
			.box-3 {
				background-color: $box-3-color;
			}
			.box-4 {
				background-color: $box-4-color;
			}
		}
	}
	.clear {
		clear: both;
	}
}
