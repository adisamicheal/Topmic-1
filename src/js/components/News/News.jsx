import React, { Component } from "react";

export default class News extends Component {
  componentDidMount() {}
  render() {
    return (
      <div className="News">
        <div className="news-Box">
          <div className="uper-box">
            <h2>Latest News</h2>
            <p>
              {" "}
              Cultural day .... Students from different cultural backgrounds came
              together to showcase their traditional costumes, food, music, and
              dance.
              <br />
              <span>03.28.18 </span>
            </p>
          </div>

          <div className="lower-box">
            <div className="des-box">
              <h3>Education Blogs</h3>
            </div>
            <div className="des-box box-3">
              <h3>Topmic Gallery</h3>
            </div>
          </div>
        </div>

        <div className="clear" />
      </div>
    );
  }
}
