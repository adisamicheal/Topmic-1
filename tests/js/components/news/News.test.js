import React from 'react';
import News from '../../../../src/js/components/News/News.jsx';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
Enzyme.configure({adapter: new Adapter()});

let wrapper = shallow(<News/>);

it('test the News Component',()=>{
	expect(wrapper).toMatchSnapshot();
});
