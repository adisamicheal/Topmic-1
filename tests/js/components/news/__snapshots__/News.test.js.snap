// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the News Component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <News />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="news-Box"
        >
          <div
            className="uper-box"
          >
            <h2>
              Latest News
            </h2>
            <p>
               
              Sed posuere consectetur est at lobor, Etiam porta sem malesuada magna..
              <br />
              <span>
                03.28.18 
              </span>
            </p>
            <p>
               
              Sed posuere consectetur est at lobor, Etiam porta sem malesuada magna..
              <br />
              <span>
                03.27.18 
              </span>
            </p>
          </div>
          <div
            className="lower-box"
          >
            <div
              className="des-box"
            >
              <h3>
                Education Blogs
              </h3>
            </div>
            <div
              className="des-box box-2"
            >
              <h3>
                Resource center
              </h3>
            </div>
            <div
              className="des-box box-3"
            >
              <h3>
                Winsome Gallery
              </h3>
            </div>
            <div
              className="des-box box-4"
            >
              <h3>
                Annual Achievers
              </h3>
            </div>
          </div>
        </div>,
        <div
          className="clear"
        />,
      ],
      "className": "News",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="uper-box"
            >
              <h2>
                Latest News
              </h2>
              <p>
                 
                Sed posuere consectetur est at lobor, Etiam porta sem malesuada magna..
                <br />
                <span>
                  03.28.18 
                </span>
              </p>
              <p>
                 
                Sed posuere consectetur est at lobor, Etiam porta sem malesuada magna..
                <br />
                <span>
                  03.27.18 
                </span>
              </p>
            </div>,
            <div
              className="lower-box"
            >
              <div
                className="des-box"
              >
                <h3>
                  Education Blogs
                </h3>
              </div>
              <div
                className="des-box box-2"
              >
                <h3>
                  Resource center
                </h3>
              </div>
              <div
                className="des-box box-3"
              >
                <h3>
                  Winsome Gallery
                </h3>
              </div>
              <div
                className="des-box box-4"
              >
                <h3>
                  Annual Achievers
                </h3>
              </div>
            </div>,
          ],
          "className": "news-Box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  Latest News
                </h2>,
                <p>
                   
                  Sed posuere consectetur est at lobor, Etiam porta sem malesuada magna..
                  <br />
                  <span>
                    03.28.18 
                  </span>
                </p>,
                <p>
                   
                  Sed posuere consectetur est at lobor, Etiam porta sem malesuada magna..
                  <br />
                  <span>
                    03.27.18 
                  </span>
                </p>,
              ],
              "className": "uper-box",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Latest News",
                },
                "ref": null,
                "rendered": "Latest News",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "Sed posuere consectetur est at lobor, Etiam porta sem malesuada magna..",
                    <br />,
                    <span>
                      03.28.18 
                    </span>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "Sed posuere consectetur est at lobor, Etiam porta sem malesuada magna..",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "03.28.18 ",
                    },
                    "ref": null,
                    "rendered": "03.28.18 ",
                    "type": "span",
                  },
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "Sed posuere consectetur est at lobor, Etiam porta sem malesuada magna..",
                    <br />,
                    <span>
                      03.27.18 
                    </span>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "Sed posuere consectetur est at lobor, Etiam porta sem malesuada magna..",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "03.27.18 ",
                    },
                    "ref": null,
                    "rendered": "03.27.18 ",
                    "type": "span",
                  },
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="des-box"
                >
                  <h3>
                    Education Blogs
                  </h3>
                </div>,
                <div
                  className="des-box box-2"
                >
                  <h3>
                    Resource center
                  </h3>
                </div>,
                <div
                  className="des-box box-3"
                >
                  <h3>
                    Winsome Gallery
                  </h3>
                </div>,
                <div
                  className="des-box box-4"
                >
                  <h3>
                    Annual Achievers
                  </h3>
                </div>,
              ],
              "className": "lower-box",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h3>
                    Education Blogs
                  </h3>,
                  "className": "des-box",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Education Blogs",
                  },
                  "ref": null,
                  "rendered": "Education Blogs",
                  "type": "h3",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h3>
                    Resource center
                  </h3>,
                  "className": "des-box box-2",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Resource center",
                  },
                  "ref": null,
                  "rendered": "Resource center",
                  "type": "h3",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h3>
                    Winsome Gallery
                  </h3>,
                  "className": "des-box box-3",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Winsome Gallery",
                  },
                  "ref": null,
                  "rendered": "Winsome Gallery",
                  "type": "h3",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h3>
                    Annual Achievers
                  </h3>,
                  "className": "des-box box-4",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Annual Achievers",
                  },
                  "ref": null,
                  "rendered": "Annual Achievers",
                  "type": "h3",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "clear",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="news-Box"
          >
            <div
              className="uper-box"
            >
              <h2>
                Latest News
              </h2>
              <p>
                 
                Sed posuere consectetur est at lobor, Etiam porta sem malesuada magna..
                <br />
                <span>
                  03.28.18 
                </span>
              </p>
              <p>
                 
                Sed posuere consectetur est at lobor, Etiam porta sem malesuada magna..
                <br />
                <span>
                  03.27.18 
                </span>
              </p>
            </div>
            <div
              className="lower-box"
            >
              <div
                className="des-box"
              >
                <h3>
                  Education Blogs
                </h3>
              </div>
              <div
                className="des-box box-2"
              >
                <h3>
                  Resource center
                </h3>
              </div>
              <div
                className="des-box box-3"
              >
                <h3>
                  Winsome Gallery
                </h3>
              </div>
              <div
                className="des-box box-4"
              >
                <h3>
                  Annual Achievers
                </h3>
              </div>
            </div>
          </div>,
          <div
            className="clear"
          />,
        ],
        "className": "News",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="uper-box"
              >
                <h2>
                  Latest News
                </h2>
                <p>
                   
                  Sed posuere consectetur est at lobor, Etiam porta sem malesuada magna..
                  <br />
                  <span>
                    03.28.18 
                  </span>
                </p>
                <p>
                   
                  Sed posuere consectetur est at lobor, Etiam porta sem malesuada magna..
                  <br />
                  <span>
                    03.27.18 
                  </span>
                </p>
              </div>,
              <div
                className="lower-box"
              >
                <div
                  className="des-box"
                >
                  <h3>
                    Education Blogs
                  </h3>
                </div>
                <div
                  className="des-box box-2"
                >
                  <h3>
                    Resource center
                  </h3>
                </div>
                <div
                  className="des-box box-3"
                >
                  <h3>
                    Winsome Gallery
                  </h3>
                </div>
                <div
                  className="des-box box-4"
                >
                  <h3>
                    Annual Achievers
                  </h3>
                </div>
              </div>,
            ],
            "className": "news-Box",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2>
                    Latest News
                  </h2>,
                  <p>
                     
                    Sed posuere consectetur est at lobor, Etiam porta sem malesuada magna..
                    <br />
                    <span>
                      03.28.18 
                    </span>
                  </p>,
                  <p>
                     
                    Sed posuere consectetur est at lobor, Etiam porta sem malesuada magna..
                    <br />
                    <span>
                      03.27.18 
                    </span>
                  </p>,
                ],
                "className": "uper-box",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Latest News",
                  },
                  "ref": null,
                  "rendered": "Latest News",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "Sed posuere consectetur est at lobor, Etiam porta sem malesuada magna..",
                      <br />,
                      <span>
                        03.28.18 
                      </span>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "Sed posuere consectetur est at lobor, Etiam porta sem malesuada magna..",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "03.28.18 ",
                      },
                      "ref": null,
                      "rendered": "03.28.18 ",
                      "type": "span",
                    },
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "Sed posuere consectetur est at lobor, Etiam porta sem malesuada magna..",
                      <br />,
                      <span>
                        03.27.18 
                      </span>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "Sed posuere consectetur est at lobor, Etiam porta sem malesuada magna..",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "03.27.18 ",
                      },
                      "ref": null,
                      "rendered": "03.27.18 ",
                      "type": "span",
                    },
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="des-box"
                  >
                    <h3>
                      Education Blogs
                    </h3>
                  </div>,
                  <div
                    className="des-box box-2"
                  >
                    <h3>
                      Resource center
                    </h3>
                  </div>,
                  <div
                    className="des-box box-3"
                  >
                    <h3>
                      Winsome Gallery
                    </h3>
                  </div>,
                  <div
                    className="des-box box-4"
                  >
                    <h3>
                      Annual Achievers
                    </h3>
                  </div>,
                ],
                "className": "lower-box",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h3>
                      Education Blogs
                    </h3>,
                    "className": "des-box",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Education Blogs",
                    },
                    "ref": null,
                    "rendered": "Education Blogs",
                    "type": "h3",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h3>
                      Resource center
                    </h3>,
                    "className": "des-box box-2",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Resource center",
                    },
                    "ref": null,
                    "rendered": "Resource center",
                    "type": "h3",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h3>
                      Winsome Gallery
                    </h3>,
                    "className": "des-box box-3",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Winsome Gallery",
                    },
                    "ref": null,
                    "rendered": "Winsome Gallery",
                    "type": "h3",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h3>
                      Annual Achievers
                    </h3>,
                    "className": "des-box box-4",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Annual Achievers",
                    },
                    "ref": null,
                    "rendered": "Annual Achievers",
                    "type": "h3",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "clear",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
